lib/train/transports/ssh.rb

```diff
@@ -85,6 +88,14 @@ def connection(state = {}, &block)
 
     private
 
+    def reusable_connection?(conn_opts)
+      return false unless @connection_options
+
+      current_options = @connection_options.dup
```

Comment:
If `@connection_options` is not mutated, why is a `.dup` call necessary here?

Reply:
Just followed the suggestion here, assuming there was some Ruby knowledge beyond me. If it's not needed, I am happy to drop it.

Reply:
I asked for it on an earlier commit where we were calling `reject!` on `current_options` - it can be removed now.